pkg/reconciler/workload/resource/resource_reconcile.go

```diff
@@ -68,6 +68,22 @@ func (c *Controller) reconcileResource(ctx context.Context, lclusterName logical
 
 	annotationPatch, labelPatch := computePlacement(ns, obj)
 
+	// If the object DeletionTimestamp is set, we should set all locations deletion timestamps annotations to the same value.
+	if obj.GetDeletionTimestamp() != nil {
+		klog.V(2).Infof("Resource is being deleted; setting the deletion per locations timestamps for %s|%s/%s", lclusterName, obj.GetNamespace(), obj.GetName())
+		objAnnotations := obj.GetAnnotations()
+		objLocations, _ := locations(objAnnotations, obj.GetLabels(), false)
+
+		if annotationPatch == nil {
+			annotationPatch = make(map[string]interface{})
+		}
+		for location := range objLocations {
+			if val, ok := objAnnotations[workloadv1alpha1.InternalClusterDeletionTimestampAnnotationPrefix+location]; !ok || val == "" {
+				annotationPatch[workloadv1alpha1.InternalClusterDeletionTimestampAnnotationPrefix+location] = obj.GetDeletionTimestamp().String()
+			}
+		}
+	}
```

Comment:
can we move that into `annotationPatch, labelPatch := propagateDeletionTimestamp(obj, annotationPath, labelPatch)` ?

Reply:
and add a unit test for that func

Reply:
question: can you annotate an object that is going to be deleted?